BaseClasses.py

```diff
@@ -1254,23 +1250,31 @@ def parse_data(self):
 
         self.locations = OrderedDict()
         listed_locations = set()
+        lw_locations = []
```

Comment:
This entire function is specific for LTTP handling as far as I can tell and out of scope for this PR. Planning to revisit it after moving each of the options it parses here.

Reply:
I do agree with treble on this one: getting rid of the lttp details in BaseClasses / other worlds is separate from getting rid of the handling here. Rewriting the whole function is kind of a bigger task and future change to this can be made without breaking the API once this is merged.